input-output-hk/cardano-sl
# [CBR-310] Reject HTTP/2.0 requests using Wai middleware

## Description

For some requests (involving streaming), HTTP/2.0 causes the connection to drop and result in very weird messages that can be hard to debug for clients. This will systematically reject HTTP/2.0 request with a proper error message suggesting to use HTTP/1.1 as a protocol.

## Linked issue

[[CBR-310]](https://iohk.myjetbrains.com/youtrack/issue/CBR-310)

## Type of change


## Developer checklist


## Testing checklist


## QA Steps

- Integration tests should run and capture the new behavior
- If ever:

```
$ curl --http2 -k --cert scripts/tls-files/client.pem https://localhost:8090/api/v1/wallets | json_pp 
{
   "message" : "NotSupportedError",
   "status" : "error",
   "diagnostic" : "HTTP/2.0 protocol isn't yet supported by the API. Please configure your client to rely on HTTP/1.1 instead."
}
```


## Screenshots (if available)
